src/core/Particle.hpp

```diff
@@ -268,7 +268,7 @@ struct ParticlePosition {
 
 #ifdef BOND_CONSTRAINT
   /** particle position at the previous time step (RATTLE algorithm) */
-  Utils::Vector3d p_old = {0., 0., 0.};
+  Utils::Vector3d p_last_timestep = {0., 0., 0.};
```

Comment:
From what I understand, this member cannot be moved to `ParticleLocal` (7713989ad856a) because we need to communicate it on ghosts, correct?

Reply:
The problem is that the correction term is calculated using `p.r.p` as well as `p.r.p_old` and both of them have to be present on the ghost particles. If my understanding is correct this means that i can't use `ParticleLocal` for that.
https://github.com/espressomd/espresso/blob/294f4905578f26b08826e463426b61688fc4a18e/src/core/rattle.cpp#L82-L91

Reply:
I might be missing something here, but i should be able to get this value from the start of the integration and store it locally because it is only a copy of the `p.r.p` value... Let me have a look at that tomorrow with a fresh mind.

Reply:
If my understand is correct it is not possible to store `p_last_timestep` in `ParticleLocal` because of this call which is resorting the cellstructure.
https://github.com/espressomd/espresso/blob/1e37bedd09f9569a45a4129db375e624abdbd157/src/core/rattle.cpp#L134

If we need to get rid of it from the `ParticlePosition` struct i guess a new one needs to be introduced because highjacking the `ParticleRattle` struct is probably not a good idea since a reduction is operating on it and a different communication-operation is expected for `p_last_timestep`.